www/js/controllers.js

```diff
@@ -1,56 +1,131 @@
+    });
+
+    // Triggered in the login modal to close it
+    $scope.closeLogin = function () {
+      $scope.loginmodal.hide();
+    };
+
+    $scope.closeSignup = function () {
+      $scope.signupmodal.hide();
+    };
+
+    // Open the login modal
+    $scope.login = function () {
+      $scope.loginmodal.show();
+    };
 
-  // Triggered in the login modal to close it
-  $scope.closeLogin = function() {
-    $scope.modal.hide();
-  };
+    $scope.signup = function () {
+      $scope.signupmodal.show();
+    };
 
-  // Open the login modal
-  $scope.login = function() {
-    $scope.modal.show();
-  };
+    $scope.doSignup = function () {
+      $ionicLoading.show({
+        template: 'Signing up...'
+      });
+      $auth.submitRegistration($scope.signupData)
+        .then(function (resp) {
+          $ionicLoading.hide();
+          $scope.closeSignup();
+          $ionicLoading.show({
+            template: 'Successfully signed up!',
+            duration: 1000
+          });
+        })
+        .catch(function (response) {
+          $ionicLoading.hide();
+          $scope.errorMessage = response;
+          $scope.showAlert('Failure', response.data.errors.full_messages);
+        });
+    };
+    // Perform the login action when the user submits the login form
+    $scope.doLogin = function () {
+      $ionicLoading.show({
+        template: 'Logging in...'
+      });
+      $auth.submitLogin($scope.loginData)
+        .then(function (resp) {
+          $ionicLoading.hide();
+          $scope.closeLogin();
+          $ionicLoading.show({
+            template: 'Successfully logged in!',
+            duration: 1000
+          });
+        })
+        .catch(function (response) {
+          $ionicLoading.hide();
+          $scope.errorMessage = response;
+          $scope.showAlert('Failure', response.data.errors.full_messages);
+        });
+    };
 
-  // Perform the login action when the user submits the login form
-  $scope.doLogin = function() {
-    console.log('Doing login', $scope.loginData);
+
+    $scope.showAlert = function (message, content) {
+      var alertPopup = $ionicPopup.alert({
+        title: message,
+        template: content
+      });
+      alertPopup.then(function (res) {
+
+      });
+    };
 
     // Simulate a login delay. Remove this and replace with your login
     // code if using a login system
-    $timeout(function() {
+    $timeout(function () {
       $scope.closeLogin();
     }, 1000);
-  };
-})
-
-.controller('PlaylistsCtrl', function($scope) {
-  $scope.playlists = [
-    { title: 'Reggae', id: 1 },
-    { title: 'Chill', id: 2 },
-    { title: 'Dubstep', id: 3 },
-    { title: 'Indie', id: 4 },
-    { title: 'Rap', id: 5 },
-    { title: 'Cowbell', id: 6 }
-  ];
-})
-
-.controller('PlaylistCtrl', function($scope, $stateParams) {
-});
+
+    $rootScope.$on('auth:login-success', function (ev, user) {
+      $scope.currentUser = angular.extend(user, $auth.retrieveData('auth_headers'));
+    });
```

Comment:
I assumed this was an AUTH event. Like when the login is successful, currentUser becomes the user which has just logged in. It is basically a way for the app to know which user is currently using it. And i think that when there's a new request (for example add to order where user has to be logged in), the app will send that header with the request, so the api knows which user's order to handle. Am I right? @tochman